### ISSUE TITLE
Setting oauth2.pkce.enforced to true does not work

### ISSUE THREAD

**Issue by dteoh** [ory/hydra#1747]
**Describe the bug**

Setting oauth2.pkce.enforced through `OAUTH2_PKCE_ENFORCED=true` or `OAUTH2_PKCE_ENFORCED=1` does not result in PKCE being mandatory.

**Reproducing the bug**

1. Configure `oauth2.pkce.enforced` to `true` either through the YAML config file (https://www.ory.sh/docs/hydra/configuration) or environment variable `OAUTH2_PKCE_ENFORCED`. I have tried both `true` and `1` when using the environment variable.
2. Start hydra. I use docker compose to run "oryd/hydra:v1.3.2".
3. Create authorization grant code flow, but omit PKCE parameters. eg.

`POST /oauth2/auth?audience=http%3A%2F%2Flocalhost%3A5555&client_id=localhost_demo_app&redirect_uri=http%3A%2F%2Flocalhost%3A5000%2Fcallback&response_type=code&scope=openid+offline&state=i6qNJyDPDrIbTEucj4Sc%2Bs52OIVjjxtVZ0Ez7s%2FZJ116%2FIeJibtD1A8iii%2B7ozS%2FqNlL5MTGgIwivs2LQBEjOg%3D%3D&nonce=4c01b5637b3979eeaaab1e282e6ac095497c4e45cb70dbb3fb56af45740b448b` (code_challenge and code_challenge_method parameters used by PKCE are missing).

4. Request succeeds. Hydra issues a 302.

*Server configuration*

```
version: '3'
services:
  postgresd:
    image: postgres:9.6
    ports:
      - "5432"
    environment:
      POSTGRES_USER: hydra
      POSTGRES_PASSWORD: secret
      POSTGRES_DB: hydra
  hydra-migrate:
    image: oryd/hydra:v1.3.2
    environment:
      DSN: "postgres://hydra:secret@postgresd:5432/hydra?sslmode=disable&max_conns=20&max_idle_conns=4"
    command: migrate sql -e --yes
    restart: on-failure
  hydra:
    image: oryd/hydra:v1.3.2
    ports:
      - "9000:4444" # Public port
      - "9001:4445" # Admin port
    depends_on:
      - hydra-migrate
    environment:
      SECRETS_SYSTEM: secret
      DSN: "postgres://hydra:secret@postgresd:5432/hydra?sslmode=disable&max_conns=20&max_idle_conns=4"
      URLS_SELF_ISSUER: "https://localhost:9000"
      URLS_CONSENT: "http://localhost:5000/hydra/consent/new"
      URLS_LOGIN: "http://localhost:5000/hydra/login/new"
      URLS_LOGOUT: "http://localhost:5000/hydra/logout/new"
      URLS_POST_LOGOUT_REDIRECT: "http://localhost:5000/hydra/logout"
      OAUTH2_ERROR_URL: "http://localhost:5000/hydra/error"
      OAUTH2_PKCE_ENFORCED: "true"
      SERVE_PUBLIC_CORS_ENABLED: "true"
      SERVE_PUBLIC_CORS_ALLOWED_ORIGINS: "*"
```
**Expected behavior**

Request does not succeed. Error response is created as per the RFC:

https://tools.ietf.org/html/rfc7636#section-4.4.1

**Environment**

* Version: oryd/hydra:v1.3.2
* Environment: Docker for Mac 2.2.0.3

**Comment by aluck1** [comment]
When is this going to be fixed?

**Comment by aeneasr** [comment]
You can give it a try!

**Comment by aeneasr** [comment]
This was fixed in 1.4.8
